autonomy/cli/scaffold_fsm_templates/tests.py

```diff
@@ -0,0 +1,324 @@
+        \"\"\"Base test class for {FSMName} rounds.\"\"\"
+
+        synchronized_data: SynchronizedData
+        _synchronized_data_class = SynchronizedData
+        _event_class = Event
+
+        def run_test(self, test_case: RoundTestCase, **kwargs) -> None:
+            \"\"\"Run the test\"\"\"
+
+            self.synchronized_data.update(**test_case.initial_data)
+
+            test_round = self.round_class(
+                synchronized_data=self.synchronized_data,
+                consensus_params=self.consensus_params,
+            )
+
+            self._complete_run(
+                self._test_round(
+                    test_round=test_round,
+                    round_payloads=test_case.payloads,
+                    synchronized_data_update_fn=lambda sync_data, _: sync_data.update(**test_case.final_data),
+                    synchronized_data_attr_checks=test_case.synchronized_data_attr_checks,
+                    exit_event=test_case.event,
+                    **kwargs,  # varies per BaseRoundTestClass child
+                )
+            )
+
+    """
+
+    TEST_ROUND_CLS = """\
+    class Test{RoundCls}(Base{FSMName}RoundTest):
+        \"\"\"Tests for {RoundCls}.\"\"\"
+
+        round_class = {RoundCls}
+
+        # TODO: provide test cases
+        @pytest.mark.parametrize("test_case, kwargs", [])
+        def test_run(self, test_case: RoundTestCase, **kwargs: Any) -> None:
+            \"\"\"Run tests.\"\"\"
+
+            self.run_test(test_case, **kwargs)
+
+    """
+
+
+@dataclass
+class TEST_BEHAVIOURS:
+    """Template for test_behaviours.py"""
+
+    FILENAME = "test_behaviours.py"
+
+    HEADER = """\
+    \"\"\"This package contains round behaviours of {AbciApp}.\"\"\"
+
+    from pathlib import Path
+    from typing import Any, Dict, Hashable, Optional, Type
+    from dataclasses import dataclass
+
+    import pytest
+
+    from packages.valory.skills.abstract_round_abci.base import AbciAppDB
+    from packages.valory.skills.abstract_round_abci.behaviours import (
+        AbstractRoundBehaviour,
+        BaseBehaviour,
+        make_degenerate_behaviour,
+    )
+    from packages.{author}.skills.{skill_name}.behaviours import (
+        {FSMName}BaseBehaviour,
+        {FSMName}RoundBehaviour,
+        {behaviours},
+    )
+    from packages.{author}.skills.{skill_name}.rounds import (
+        SynchronizedData,
+        DegenerateRound,
+        Event,
+        {AbciApp},
+        {all_rounds},
+    )
+
+    from packages.valory.skills.abstract_round_abci.test_tools.base import (
+        FSMBehaviourBaseCase,
+    )
+
+
+    @dataclass
+    class BehaviourTestCase:
+        \"\"\"BehaviourTestCase\"\"\"
+
+        initial_data: Dict[str, Hashable]
+        event: Event
+
+    """
+
+    BASE_BEHAVIOUR_TEST_CLS = """\
+    class Base{FSMName}Test(FSMBehaviourBaseCase):
+        \"\"\"Base test case.\"\"\"
+
+        path_to_skill = Path(__file__).parent.parent
+
+        behaviour: {FSMName}RoundBehaviour
```

Comment:
The name is quite confusing I think, which is what lead to the mistake below.

Reply:
introduced a property with a doc string for emphasize
https://github.com/valory-xyz/open-autonomy/blob/45e3946e73af9e5b13c113c9e8c276bd86b092f9/autonomy/cli/scaffold_fsm_templates/tests.py#L182-L186